### PR TITLE
executor: show index column length if necessary in show create table statement

### LINKED ISSUE #5384: "show create table" doesn't show key length of index for TEXT columns
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
```sql
create table wide(a text, key idx_wide_a(a(10)));
show create table wide;
```

2. What did you expect to see?
```sql
mysql> show create table wide;
+-------+-----------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                        |
+-------+-----------------------------------------------------------------------------------------------------+
| wide  | CREATE TABLE `wide` (
  `a` text,
  KEY `idx_wide_a` (`a`(10))
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+-------+-----------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

3. What did you see instead?
```sql
MySQL [gin]> show create table wide;
+-------+-------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                  |
+-------+-------------------------------------------------------------------------------------------------------------------------------+
| wide  | CREATE TABLE `wide` (
  `a` text DEFAULT NULL,
  KEY `idx_wide_a` (`a`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
+-------+-------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

4. What version of TiDB are you using (`tidb-server -V`)?
```
Release Version: v1.0.3-4-ga5f7fc9
Git Commit Hash: a5f7fc942f05153de4b2e483951f745050b3768c
Git Branch: release-1.0
UTC Build Time: 2017-12-06 11:01:37
```

### PR DESCRIPTION
**Pull request by jackysp**
Fix #5384 . PTAL @shenli @coocood

### CODE REVIEW

File: executor/show.go [hunk 1]
```diff
@@ -532,9 +533,13 @@ func (e *ShowExec) fetchShowCreateTable() error {
...
 		for _, c := range idxInfo.Columns {
-			cols = append(cols, c.Name.O)
+			colInfo := fmt.Sprintf("`%s`", c.Name.String())
+			if c.Length != types.UnspecifiedLength {
+				colInfo = fmt.Sprintf("%s(%s)", colInfo, strconv.Itoa(c.Length))
+			}
+			cols = append(cols, colInfo)
 		}
-		buf.WriteString(fmt.Sprintf("(`%s`)", strings.Join(cols, "`,`")))
+		buf.WriteString(fmt.Sprintf("(%s)", strings.Join(cols, "`,`")))
```
Comment (zz-jason) [inline executor/show.go]: no need to call `fmt.Sprintf()`

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@jackysp Some of cases in mysql-test need to be updated.

**Comment by shenli** [comment]
@jackysp Please check pick it to release-1.0 .

**Pull request merged**